### DISCUSSION TITLE
Why are order totals not saved in database?

### DISCUSSION THREAD

**Discussion by khanh-to-niteco** [medusajs/medusa#7002] [Q&A]
Hi Medusa team,

We are using Medusa 1.17.4, there are fields like `order.total`, `order.subtotal`,.. that are not saved to database. They are recalculated by `orderService.decorateTotals` method. Several questions arise:

- Do we expect these totals to change after the order was created or was last edited? For example: if we create an order 1 year ago, and there has been no change to the order, should we expect the totals to remain the same?
- If we expect the totals to remain the same when there is no edit, should we just store the totals and avoid recalculation (to avoid bugs)?
- Other issues caused by this are related to performance: in order to aggregrate on these totals, we cannot just aggregrate easily with a query to the database, we need to load the orders to aggregate on server side. It causes performance issues (also memory usage) when there are many orders.
- Cannot query orders by order totals.
- .....

Thanks,
Khanh

**Comment by srindom** [comment]
Hi Khanh - your inputs are good and in v2 we are persisting totals.
